packages/powersync_attachments_stream/lib/src/utils/mutex.dart

```diff
@@ -0,0 +1,33 @@
+// lib/src/utils/mutex.dart
+
+import 'dart:async';
+
+class AsyncMutex {
```

Comment:
I think there's a mutex implementation exposed by `package:sqlite_async` which we could re-use here.